nuxeo-features/rest-api/nuxeo-rest-api-server/src/main/java/org/nuxeo/ecm/restapi/server/jaxrs/BatchUploadObject.java

```diff
@@ -292,10 +307,14 @@ public Object execute(@PathParam(REQUEST_BATCH_ID) String batchId, @PathParam(RE
 
     protected Object executeBatch(String batchId, String fileIdx, String operationId, HttpServletRequest request,
             ExecutionRequest xreq) throws UnsupportedEncodingException {
-        RequestContext.getActiveContext(request).addRequestCleanupHandler(req -> {
-            BatchManager bm = Framework.getService(BatchManager.class);
-            bm.clean(batchId);
-        });
+
+        if (!Boolean.parseBoolean(
+            RequestContext.getActiveContext(request).getRequest().getHeader(BatchManagerConstants.NO_DROP_FLAG))) {
+            RequestContext.getActiveContext(request).addRequestCleanupHandler(req -> {
+                BatchManager bm = Framework.getService(BatchManager.class);
+                bm.clean(batchId);
+            });
+        }
```

Comment:
For comparison S3 has an "abortMultipartUploads" API that you can call, it aborts all batch uploads older than a given date. We call it at Nuxeo startup to abort all uploads older than one day. We could have a nightly scheduled task that does the same for our batch uploads. Do we store the date where the uploaded part was created? If not we should.